doc/book/app_server/luajit_memprof.rst

```diff
@@ -0,0 +1,524 @@
+
+..  code-block:: console
+
+    ALLOCATIONS
+    @format_concat.lua:3, line 4: 10000     284411  0
+
+    REALLOCATIONS
+
+    DEALLOCATIONS
+    INTERNAL: 10000 0       218905
+            Overrides:
+                    @format_concat.lua:3, line 4
+
+    @format_concat.lua:3, line 4: 1 0       32768
+
+
+**Q**: But what will change if the JIT compilation is enabled?
+
+**A**: Let's comment the first line of the code, namely, ``jit.off()`` and run
+the profiler . Now, there are only 56 allocations in the report, and all other
+allocations are JIT-related (see also the related
+`dev issue <https://github.com/tarantool/tarantool/issues/5679>`_):
+
+..  code-block:: console
+
+    ALLOCATIONS
+    @format_concat.lua:3, line 4: 56        1112    0
+    @format_concat.lua:0, line 0: 4 640     0
+    INTERNAL: 2     382     0
+
+    REALLOCATIONS
+
+    DEALLOCATIONS
+    INTERNAL: 58    0       1164
+            Overrides:
+                    @format_concat.lua:3, line 4
+                    INTERNAL
+
+This happens because a trace is compiled after 56 iterations, and the
+JIT-compiler removed the unused ``c`` variable  from the trace, and, therefore,
+the dead code of the ``concat()`` function is eliminated.
+
+Next, let's profile only the ``format()`` function with JIT enabled.
+For that, keep the lines 1 and 23 commented (``jit.off()`` and
+``local c = concat(i)`` respectively), uncomment the line 22
+(``local f = format(i)``), and run the profiler.
+
+The profiler's output is the following:
+
+..  code-block:: console
+
+    ALLOCATIONS
+    @format_concat.lua:8, line 9: 19998     624322  0
+    INTERNAL: 4     66824   0
+    @format_concat.lua:0, line 0: 4 640     0
+
+    REALLOCATIONS
+
+    DEALLOCATIONS
+    INTERNAL: 19999 0       559072
+            Overrides:
+                    @format_concat.lua:0, line 0
+                    @format_concat.lua:8, line 9
+
+    @format_concat.lua:8, line 9: 2 0       98304
+            Overrides:
+                    @format_concat.lua:8, line 9
+
+**Q**: Why is there so many allocations in comparison to the ``concat()`` function?
+
+**A**: The answer is simple: the ``string.format()`` function with the ``%s``
+identifier is not yet compiled via LuaJIT. So, a trace can't be recorded and
+the compiler doesn't perform the corresponding optimizations.
+
+If we change the ``format()`` function in the following way
+
+..  code-block:: lua
+
+    local function format(a)
+      local nstr = string.format("%sa", tostring(a))
+      return nstr
+    end
+
+the profiling report becomes much prettier:
+
+..  code-block:: console
+
+    ALLOCATIONS
+    @format_concat.lua:8, line 9: 110       2131    0
+    @format_concat.lua:0, line 0: 4 640     0
+    INTERNAL: 3     1148    0
+
+    REALLOCATIONS
+
+    DEALLOCATIONS
+    INTERNAL: 113   0       2469
+            Overrides:
+                    @format_concat.lua:0, line 0
+                    @format_concat.lua:8, line 9
+                    INTERNAL
```

Comment:
With the additional first line (see the comment above) the report looks like the following:
```
ALLOCATIONS
@format_concat.lua:9, line 10: 110       2131    0
@format_concat.lua:0, line 0: 4 640     0
INTERNAL: 3     1148    0

REALLOCATIONS

DEALLOCATIONS
INTERNAL: 113   0       2469
        Overrides:
                @format_concat.lua:0, line 0
                @format_concat.lua:9, line 10
                INTERNAL
```

Reply:
done